### ISSUE TITLE
Don't set keys of undefined values in JSON output

### ISSUE THREAD

**Issue by rdev-software** [stephenh/ts-proto#98]
when using `--ts_proto_opt=useOptionals=true`

result is correctly using optionals instead of `undefined` however methods still use it for assigning result instead of not adding them at all.

ie. when useOptionals = true resulting JSON SHOULD NOT contain keys at all as those are OPTIONAL not UNDEFINED

**Comment by stephenh** [comment]
I'm not really following; my initial assertion is that `useOptional`s should not change what the JSON encoding looks like, b/c the JSON encoding spec is defined by protobuf3, which doesn't know about TypeScript-specific features like object key optionalness.

Can you include JSON examples of like "for this proto ...paste example schema..., `toJSON` currently does ...paste JSON..., and I want it to look like ...paste JSON...".

**Comment by rdev-software** [comment]
Sure, I dont want to change protobufs spec then. However it seems strange to me having message like:

```
message User{
    string documentId = 1;
}
```

Which produces
```
{
"documentId":undefined
}
```

If field was not set, and optionals = true, it should be omitted in my opinion

**Comment by stephenh** [comment]
Ah sure, that makes sense, just for ts-proto to do that all the time actually. Would you like to submit a PR that changes this?

**Comment by stephenh** [comment]
This should be fixed in v1.32.0.
